mmeval/metrics/utils/multi_label.py

```diff
@@ -0,0 +1,169 @@
+    onehot[label] = 1
+    return onehot
+
+
+def format_data(
+    data: Union[Sequence[Union[np.ndarray, 'torch.Tensor', 'oneflow.Tensor']],
+                np.ndarray, 'torch.Tensor', 'oneflow.Tensor'],
+    num_classes: int,
+    is_onehot: bool = False
+) -> Union[np.ndarray, 'torch.Tensor', 'oneflow.Tensor']:
+    """Format data from different inputs such as prediction scores, label-
+    format data and one-hot encodings into the same output shape of `(N,
+    num_classes)`.
+
+    Args:
+        data (Union[Sequence[np.ndarray, 'torch.Tensor', 'oneflow.Tensor'],
+        np.ndarray, 'torch.Tensor', 'oneflow.Tensor']):
+            The input data of prediction or labels.
+        num_classes (int): The number of classes.
+        is_onehot (bool): Whether the data is one-hot encodings.
+
+    Return:
+        torch.Tensor or oneflow.Tensor or np.ndarray:
+        One-hot encodings or predict scores.
+    """
+    if torch and isinstance(data[0], torch.Tensor):
+        stack_func = torch.stack
+    elif flow and isinstance(data[0], flow.Tensor):
+        stack_func = flow.stack
+    elif isinstance(data[0], (np.ndarray, np.number)):
+        stack_func = np.stack
+    else:
+        raise NotImplementedError(f'Data type of {type(data[0])}'
+                                  'is not supported.')
+
+    try:
+        # try stack scores or one-hot indices directly
+        formated_data = stack_func(data)
+        # all assertions below is to find labels that are
+        # raw indices which should be caught in exception
+        # to convert to one-hot indices.
+        #
+        # 1. all raw indices has only 1 dims
+        assert formated_data.ndim == 2
+        # 2. all raw indices has the same dims
+        assert formated_data.shape[1] == num_classes
+        # 3. all raw indices has the same dims as num_classes
+        # then max indices should greater than 1 for num_classes > 2
+        assert formated_data.max() <= 1
+        # 4. corner case, num_classes=2, then one-hot indices
+        # and raw indices are undistinguishable, for instance:
+        #   [[0, 1], [0, 1]] can be one-hot indices of 2 positives
+        #   or raw indices of 4 positives.
+        # Extra induction is needed.
+        if num_classes == 2:
+            warnings.warn('Ambiguous data detected, reckoned as scores'
+                          ' or label-format data as defaults. Please set '
+                          'parms related to `is_onehot` if use one-hot '
+                          'encoding data to compute metrics.')
+            assert is_onehot
+        is_onehot = True
+    # Error corresponds to np, torch, oneflow, stack_func respectively
+    except (ValueError, RuntimeError, AssertionError):
+        is_onehot = False
+
+    if not is_onehot:
+        # convert label-format inputs to one-hot encodings
+        formated_data = stack_func(
+            [label_to_onehot(sample, num_classes) for sample in data])
```

Comment:
I still think this piece of code weird. `Try ... Except ...` should be used to handle `exceptions`, not to realize a common control flow. This whole piece of code should be converted to a `if-else-statement`.